qiskit_experiments/data_processing/data_processor.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class that ties together actions on the data."""
```

Comment:
It would be good to document the expected workflow for using a `DataProcessor`, perhaps in the top level docstring here. Here is what I think is important to know:

A `DataProcessor` defines a sequence of operations to perform on experimental data. The `DataProcessor.format_data()` method applies this sequence its input argument. A `DataProcessor` is created with a list of `DataAction` objects. Each `DataAction` specifies a set of `node_inputs` that it accepts and a `__node_output__` that it provides. The `__node_output__` of each `DataAction` must be contained in the `node_inputs` of the following `DataAction` in the `DataProcessor`'s list. `DataProcessor.format_data()` usually takes in one entry from the `data` property of an `ExperimentData` object (i.e. a `dict` containing `metadata` and `memory` keys and possibly `counts`, like the `Result.data` property) and produces a new `dict` containing the formatted data. The data passed to `DataProcessor.format_data()` is passed to the first `DataAction` and the output is passed on in turn to each `DataAction`. `DataProcessor.format_data()` returns the data produced by the last `DataAction`.

Reply:
Done in 5f7c082